recipes/aws-cdi-sdk/all/conanfile.py

```diff
@@ -0,0 +1,93 @@
+from enum import auto
+from conans import ConanFile, AutoToolsBuildEnvironment, tools, CMake
+from conans.errors import ConanInvalidConfiguration
+import os
+
+required_conan_version = ">=1.35.0"
+
+class AwsCdiSdkConan(ConanFile):
+    name = "aws-cdi-sdk"
+    description = "AWS Cloud Digital Interface (CDI) SDK"
+    topics = ("aws", "communication", "framework", "service")
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/aws/aws-cdi-sdk"
+    license = "BSD-2-Clause"
+    settings = "os", "arch", "compiler", "build_type"
+    exports_sources = ["CMakeLists.txt", "patches/**"]
+    generators = "cmake", "cmake_find_package"
+    requires = "aws-libfabric/1.9.1amzncdi1.0", "aws-sdk-cpp/1.8.130"
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    _autotools = None
+    _cmake = None
+
+    def configure(self):
+        self.options["aws-sdk-cpp"].monitoring = True
+        self.options["aws-sdk-cpp"].shared = True
+        self.options["aws-libfabric"].shared = True
+
+
+    def validate(self):
+        if self.settings.os != "Linux":
+            raise ConanInvalidConfiguration("The aws-cdi-sdk con only be built on Linux.")
+        elif (self.settings.compiler == "gcc"
+                and tools.Version(self.settings.compiler.version) < "6.0"):
+            raise ConanInvalidConfiguration("""Doesn't support gcc5 / shared.
+            See https://github.com/conan-io/conan-center-index/pull/4401#issuecomment-802631744""")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version],
+                  destination=self._source_subfolder, strip_root=True)
+
+    def _configure_autotools(self):
+        if self._autotools:
+            return self._autotools
+        self._autotools = AutoToolsBuildEnvironment(self)
+        return self._autotools
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+        self._cmake = CMake(self)
+        self._cmake.definitions["CPP_STANDARD"] = self._cmake.definitions.get("CONAN_CMAKE_CXX_STANDARD", "11")
+        self._cmake.configure()
+        return self._cmake
+
+    def build(self):        
+        for patch in self.conan_data.get("patches", {}).get(self.version, []):
+            tools.patch(**patch)
+
+        # build aws-cpp-sdk-cdi
+        cmake = self._configure_cmake()
+        cmake.build()
+        cmake.install()
```

Comment:
This should be called in the package method

Reply:
see below